uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/directives/leave-request-popup.js

```diff
@@ -0,0 +1,44 @@
+define([
+  'leave-absences/shared/modules/directives',
+  'leave-absences/shared/controllers/leave-request-popup-controller',
+], function (directives) {
+  'use strict';
+
+  directives.directive('leaveRequestPopup', ['$log', '$uibModal', 'shared-settings', 'DateFormat',
+    function ($log, $modal, settings, DateFormat) {
+      $log.debug('leaveRequestPopup');
+
+      return {
+        scope: {
+          contactId: '<'
+        },
+        restrict: 'EA',
+        link: function (scope, element, attrs, ctrl) {
+          $log.debug('link');
+
+          element.on('click', function (event) {
+            //to set HR_settings DateFormat
+            DateFormat.getDateFormat().then(function (result) {
+              scope.openComponentModal = function () {
+                $modal.open({
+                  templateUrl: settings.pathTpl + 'directives/leave-request-popup.html',
+                  animation: scope.animationsEnabled,
+                  controller: 'LeaveRequestPopupCtrl',
+                  controllerAs: '$ctrl',
```

Comment:
from uibmodel documentation `controllerAs (Type: string, Example: ctrl) - An alternative to the controller-as syntax. Requires the controller option to be provided as well.`

controllerAs is string, not sure what you mean here?